agenda-connectors-webapp/src/main/webapp/vue-app/agenda-connectors/google-connector/agendaGoogleConnector.js

```diff
@@ -62,29 +60,29 @@ export default {
               });
           }
         };
-        const cookieSuffix = this.user && this.user.substring(0, this.user.indexOf('@'));
-        const tokenResponse = getCookie(`g_connector_oauth_${cookieSuffix}`);
-        if (tokenResponse) {
-          setCookie(`g_connector_oauth_${cookieSuffix}`, tokenResponse, 90);
-          const g_stateCookie = getCookie('g_state');
-          setCookie('g_state', g_stateCookie, 90);
-        }
-        if (refresh && tokenResponse) {
-          const response = JSON.parse(tokenResponse);
-          return refreshToken(this.CLIENT_ID, this.SECRET_KEY, response.refresh_token)
-            .then(refreshTokenResponse => {
-              if (refreshTokenResponse && refreshTokenResponse.access_token) {
-                response.access_token = refreshTokenResponse.access_token;
-                setCookie(`g_connector_oauth_${cookieSuffix}`, JSON.stringify(response), 90);
-                this.gapi.client.setToken(response);
-                this.canPush = this.cientOauth.hasGrantedAllScopes(response, this.SCOPE_WRITE);
-                resolve(response);
-              }
-            });
-        } else if (tokenResponse && this.user) {
-          const response = JSON.parse(tokenResponse);
-          this.gapi.client.setToken(response);
-          resolve(response);
+        if (refresh) {
+          return refreshToken().then(refreshTokenResponse => {
+            if (refreshTokenResponse && refreshTokenResponse.access_token) {
+              this.gapi.client.setToken(refreshTokenResponse);
+              this.canPush = this.cientOauth.hasGrantedAllScopes(refreshTokenResponse, this.SCOPE_WRITE);
+              resolve(refreshTokenResponse);
+            }
+          });
+        } else if (this.user) {
+          return getStoredToken().then(tokenResponse => {
+            if (tokenResponse && tokenResponse.access_token) {
+              this.gapi.client.setToken(tokenResponse);
+              this.canPush = this.cientOauth.hasGrantedAllScopes(tokenResponse, this.SCOPE_WRITE);
+              this.gapi.client.setToken(tokenResponse);
+              const g_stateCookie = getCookie('g_state');
+              setCookie('g_state', g_stateCookie, 90);
```

Comment:
Do we still need the g_state cookie ? What do we store into ?

Reply:
removed